src/hooks/useProfileRedirect.ts

```diff
@@ -87,29 +102,29 @@ export const useProfileRedirect = () => {
 	const [redirectionComplete, setRedirectionComplete] = useState(false);
 
 	useEffect(() => {
-		if (hasRedirected.current) {
-			return;
-		}
-
 		// Create abort controller to fetch abort
 		const controller = new AbortController();
 		const { signal } = controller;
 
 		const redirect = async () => {
 			try {
-				const redirectPath = await getRedirectPath(signal);
+				// Normalize URL path
+				const currentPath = new URL(window.location.href).pathname;
+
+				const redirectPath = await getRedirectPath(signal, currentPath);
+
```

Comment:
Why are we checking for a signal abort here?  The signal is aborted in the callback clean for useEffect, which happens after.

Reply:
I think the abort signal is for canceling the promise.

https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal